sklearn/cluster/tests/test_k_means.py

```diff
@@ -833,7 +843,7 @@ def test_kmeans_warns_less_centers_than_unique_points():
         km.fit(X)
         # only three distinct points, so only three clusters
         # can have points assigned to them
-        assert set(km.labels_) == set(range(3))
+        assert len(set(km.labels_)) == 3
```

Comment:
As discussed irl, the way kmeans_plusplus works guarantees that the clusters assigned to duplicated points should have the highest indices, making this assertion correct no matter the rng. Let's keep the addition of the global_random_seed to this test though.
```suggestion
        assert set(km.labels_) == set(range(3))
```